### PR TITLE
Cleared fields for other authentication methods when using one

### PR DESCRIPTION
**Pull request by dhruvanshus-crest**
Jira: https://jira.splunk.com/browse/ADDON-38274

- Cleared fields for other authentication methods when using one

### CODE REVIEW

File: src/main/webapp/components/BaseFormView.jsx [hunk 1]
```diff
@@ -485,6 +485,9 @@ class BaseFormView extends PureComponent {
                 this.isOAuth &&
                 (this.isSingleOauth || (this.isAuthVal && this.datadict.auth_type === 'oauth'))
             ) {
+                // clear out Basic Authentication fields before doing OAuth Authentication
+                this.datadict.username = '';
+                this.datadict.password = '';
```
Comment (harshpatel-crest) [inline src/main/webapp/components/BaseFormView.jsx:490]: This is static. We need to use `globalConfig.json` to get field names dynamically and use that. There should be a variable already defined as far as I know which tracks fields for oauth.

File: src/main/webapp/components/BaseFormView.jsx [hunk 2]
```diff
@@ -569,6 +572,9 @@ class BaseFormView extends PureComponent {
                     }
                 });
             } else {
+                // clear out OAuth Authentication fields before doing Basic Authentication
+                this.datadict.client_id = '';
+                this.datadict.client_secret = '';
```
Comment (harshpatel-crest) [inline src/main/webapp/components/BaseFormView.jsx:577]: Same goes here.

File: src/main/webapp/components/BaseFormView.jsx (hunk 1 above)
Comment (harshpatel-crest) [reply src/main/webapp/components/BaseFormView.jsx:490]: For reference:
https://github.com/splunk/addonfactory-ucc-base-ui/blob/c222d1a7bb0b13730f8e21fef051cc5cc1a48ad3/src/main/webapp/components/BaseFormView.jsx#L455-L459

Comment (dhruvanshus-crest) [reply src/main/webapp/components/BaseFormView.jsx]: Done

File: src/main/webapp/components/BaseFormView.jsx (hunk 2 above)
Comment (dhruvanshus-crest) [reply src/main/webapp/components/BaseFormView.jsx]: Okay

File: src/main/webapp/components/BaseFormView.jsx [hunk 3]
```diff
@@ -593,6 +593,17 @@ class BaseFormView extends PureComponent {
             }
         });
 
+        // clear out fields of other authentication methods when using one
```
Comment (harshpatel-crest) [inline src/main/webapp/components/BaseFormView.jsx:596]: nit: Using word `one` here does not give context to inline comment.

**Review by harshpatel-crest — APPROVED** [review]

**Pull request merged**